lib/Jsrt/JsrtDebugManager.cpp

```diff
@@ -0,0 +1,693 @@
+        JsrtDebugUtils::AddSourceLengthAndTextToObject(eventDataObject, functionBody, currentByteCodeOffset);
+
+        this->CallDebugEventCallbackForBreak(jsDiagDebugEvent, eventDataObject, scriptContext);
+    }
+}
+
+void JsrtDebugManager::ReportExceptionBreak(Js::InterpreterHaltState* haltState)
+{
+    if (this->debugEventCallback != nullptr)
+    {
+        Assert(haltState->stopType == Js::STOP_EXCEPTIONTHROW);
+
+        Js::Utf8SourceInfo* utf8SourceInfo = haltState->GetFunction()->GetUtf8SourceInfo();
+        Js::ScriptContext* scriptContext = utf8SourceInfo->GetScriptContext();
+
+        JsDiagDebugEvent jsDiagDebugEvent = JsDiagDebugEventRuntimeException;
+
+        JsrtDebugEventObject debugEventObject(scriptContext);
+
+        Js::DynamicObject* eventDataObject = debugEventObject.GetEventDataObject();
+
+        JsrtDebugUtils::AddScriptIdToObject(eventDataObject, utf8SourceInfo);
+
+        Js::FunctionBody* functionBody = haltState->topFrame->GetFunction();
+
+        Assert(functionBody != nullptr);
+        
+        int currentByteCodeOffset = haltState->topFrame->GetByteCodeOffset();
+        JsrtDebugUtils::AddLineColumnToObject(eventDataObject, functionBody, currentByteCodeOffset);
+        JsrtDebugUtils::AddSourceLengthAndTextToObject(eventDataObject, functionBody, currentByteCodeOffset);
+        JsrtDebugUtils::AddPropertyToObject(eventDataObject, JsrtDebugPropertyId::uncaught, !haltState->exceptionObject->IsFirstChanceException(), scriptContext);
+
+        Js::ResolvedObject resolvedObject;
+        resolvedObject.scriptContext = scriptContext;
+        resolvedObject.name = _u("{exception}");
+        resolvedObject.typeId = Js::TypeIds_Error;
+        resolvedObject.address = nullptr;
+        resolvedObject.obj = scriptContext->GetDebugContext()->GetProbeContainer()->GetExceptionObject();
+
+        if (resolvedObject.obj == nullptr)
+        {
+            resolvedObject.obj = resolvedObject.scriptContext->GetLibrary()->GetUndefined();
+        }
+
+        JsrtDebuggerObjectBase::CreateDebuggerObject<JsrtDebuggerObjectProperty>(this->GetDebuggerObjectsManager(), resolvedObject, scriptContext, [&](Js::Var marshaledObj)
+        {
+            JsrtDebugUtils::AddPropertyToObject(eventDataObject, JsrtDebugPropertyId::exception, marshaledObj, scriptContext);
+        });
+        
+        this->CallDebugEventCallbackForBreak(jsDiagDebugEvent, eventDataObject, scriptContext);
+    }
+}
+
+void JsrtDebugManager::HandleResume(Js::InterpreterHaltState* haltState, BREAKRESUMEACTION resumeAction)
+{
+    Js::ScriptContext* scriptContext = haltState->framePointers->Peek()->GetScriptContext();
+
+    if (resumeAction == BREAKRESUMEACTION_ABORT)
+    {
+        // In this case we need to abort script execution entirely and also stop working on any breakpoint for this engine.
+        scriptContext->GetDebugContext()->GetProbeContainer()->RemoveAllProbes();
```

Comment:
Would be good if this can move to ProbeContainer (I believe this logic duplicated from full repo).